plugin/gatsby-node.js

```diff
@@ -2,52 +2,73 @@ require("dotenv").config()
 const fetch = require("node-fetch")
 const { createNodeHelpers } = require("gatsby-node-helpers")
 const { createInterface } = require("readline")
-const { finishLastOperation, createOperation, completedOperation } = require('./operations')
-const { nodeBuilder } = require('./node-builder')
+const { finishLastOperation, createProductsOperation, createOrdersOperation, completedOperation } = require('./operations')
+const { nodeBuilder, idPattern } = require('./node-builder')
+
+module.exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    shopifyConnections: Joi.array().default([]).items(Joi.string().valid('orders'))
+  })
+}
+
+module.exports.sourceNodes = async function({ reporter, actions, createNodeId, createContentDigest }, pluginOptions) {
+  const operations = [createProductsOperation]
+  if (pluginOptions.shopifyConnections.includes('orders')) {
+    operations.push(createOrdersOperation)
+  }
 
-module.exports.sourceNodes = async function({ reporter, actions, createNodeId, createContentDigest }) {
   const nodeHelpers = createNodeHelpers({
     typePrefix: `Shopify`,
     createNodeId,
     createContentDigest,
   })
 
-  await finishLastOperation()
+  for await (const op of operations) {
+    await finishLastOperation()
 
-  const { bulkOperationRunQuery: { userErrors, bulkOperation} } = await createOperation()
+    const { bulkOperationRunQuery: { userErrors, bulkOperation} } = await op()
 
-  if (userErrors.length) {
-    reporter.panic({
-      context: {
-        sourceMessage: `Couldn't perform bulk operation`
-      }
-    }, ...userErrors)
-  }
+    if (userErrors.length) {
+      reporter.panic({
+        context: {
+          sourceMessage: `Couldn't perform bulk operation`
+        }
+      }, ...userErrors)
+    }
 
-  let resp = await completedOperation(bulkOperation.id)
+    let resp = await completedOperation(bulkOperation.id)
 
-  const results = await fetch(resp.node.url)
+    const results = await fetch(resp.node.url)
 
-  /* FIXME
-   * Getting warnings about this being experimental.
-   * We want to read the stream one line at a time, but let's make
-   * sure to do it in the recommended way.
-   */
-  const rl = createInterface({
-    input: results.body,
-    crlfDelay: Infinity,
-  })
+    /* FIXME
+    * Getting warnings about this being experimental.
+    * We want to read the stream one line at a time, but let's make
+    * sure to do it in the recommended way.
+    */
+    const rl = createInterface({
+      input: results.body,
+      crlfDelay: Infinity,
+    })
+
+    const objects = []
+    for await (const line of rl) {
```

Comment:
We have two loops here: one to parse the file line by line, and one to create nodes from each parsed object. I did this originally because the Shopify docs advise users to read the file in backwards. But this does not seem to be necessary at all for Gatsby so I might consolidate the loops.